### ISSUE TITLE
[Bugs] Build OK but can't run in ArchLinux

### ISSUE THREAD

**Issue by zxp19821005** [RenanSui/salvage#1]
As titled.
### Environment
Kernel: 6.6.16-1-lts
npm: 10.2.3
nodejs: 18.19.0
pnpm: 8.15.1

### Build Logs
```
克隆到纯仓库 '/home/zxp-endeavouros/.cache/paru/clone/salvage-git/salvage.git'...
remote: Enumerating objects: 606, done.
remote: Counting objects: 100% (324/324), done.
remote: Compressing objects: 100% (182/182), done.
remote: Total 606 (delta 179), reused 264 (delta 140), pack-reused 282
接收对象中: 100% (606/606), 1.03 MiB | 831.00 KiB/s, 完成.
处理 delta 中: 100% (281/281), 完成.
  -> 找到 salvage.sh
==> 正在验证 source 文件，使用sha256sums...
    salvage.git ... 已跳过
    salvage.sh ... 通过
==> 正在释放源码...
  -> 正在建立 salvage.git git 仓库的拷贝...
正克隆到 'salvage.git'...
完成。
==> 正在开始 pkgver()...
==> 已升级版本：salvage-git 1.5.3.r2.g846765d-1
==> 正在开始 build()...
Downloading and installing node v18.19.0...
Downloading https://nodejs.org/dist/v18.19.0/node-v18.19.0-linux-x64.tar.xz...
##################################################################################################################################################################################################### 100.0%
Computing checksum with sha256sum
Checksums matched!
Now using node v18.19.0 (npm v10.2.3)
Creating default alias: default -> 18 (-> v18.19.0)
Now using node v18.19.0 (npm v10.2.3)
Lockfile is up to date, resolution step is skipped
Packages: +625
++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
Downloading registry.npmjs.org/typescript/5.1.6: 7.15 MB/7.15 MB, done
Downloading registry.npmjs.org/app-builder-bin/4.0.0: 45.10 MB/45.10 MB, done
Progress: resolved 625, reused 0, downloaded 625, added 625, done
node_modules/.pnpm/esbuild@0.18.17/node_modules/esbuild: Running postinstall script, done in 61ms
node_modules/.pnpm/electron@24.6.5/node_modules/electron: Running postinstall script, done in 81ms

dependencies:
+ @electron-toolkit/preload 2.0.0
+ @electron-toolkit/utils 1.0.2
+ @hookform/resolvers 3.1.1
+ @radix-ui/react-label 2.0.2
+ @radix-ui/react-slot 1.0.2
+ chokidar 3.5.3
+ class-variance-authority 0.7.0
+ clsx 2.0.0
+ electron-debug 3.2.0
+ electron-log 4.4.8
+ electron-store 8.1.0
+ electron-updater 5.3.0
+ fs-extra 11.1.1
+ lucide-react 0.263.1
+ react-hook-form 7.45.2
+ sonner 0.6.2
+ tailwind-variants 0.1.13
+ zod 3.21.4

devDependencies:
+ @electron-toolkit/tsconfig 1.0.1
+ @electron/notarize 1.2.4
+ @rocketseat/eslint-config 2.1.0
+ @total-typescript/ts-reset 0.4.2
+ @types/node 18.17.1
+ @types/react 18.2.18
+ @types/react-dom 18.2.7
+ @types/string-similarity 4.0.0
+ @typescript-eslint/eslint-plugin 5.62.0
+ @typescript-eslint/parser 5.62.0
+ @vitejs/plugin-react 4.0.4
+ ajv 8.12.0
+ autoprefixer 10.4.14
+ electron 24.6.5
+ electron-builder 23.6.0
+ electron-tray-window 1.2.7
+ electron-vite 1.0.26
+ eslint 8.46.0
+ eslint-config-prettier 8.9.0
+ eslint-plugin-prettier 4.2.1
+ eslint-plugin-react 7.33.1
+ postcss 8.4.27
+ prettier 2.8.8
+ react 18.2.0
+ react-dom 18.2.0
+ tailwind-merge 1.14.0
+ tailwindcss 3.3.3
+ tailwindcss-animate 1.0.6
+ typescript 5.1.6
+ vite 4.4.7

> salvage@1.5.3 postinstall /home/zxp-endeavouros/.cache/paru/clone/salvage-git/src/salvage.git
> electron-builder install-app-deps

  • electron-builder  version=23.6.0
  • loaded configuration  file=/home/zxp-endeavouros/.cache/paru/clone/salvage-git/src/salvage.git/electron-builder.yml
┌────────────────────────────────────────────┐
│ New version of electron-builder available! │
│ Current Version: 23.6.0                    │
│ Latest Version: 24.9.1                     │
└────────────────────────────────────────────┘

Done in 25.5s

> salvage@1.5.3 build:linux /home/zxp-endeavouros/.cache/paru/clone/salvage-git/src/salvage.git
> electron-vite build && electron-builder --linux --config

vite v4.4.7 building for production...
✓ 7 modules transformed.
out/main/index.js  10.71 kB
✓ built in 243ms
vite v4.4.7 building for production...
✓ 1 modules transformed.
out/preload/index.js  1.33 kB
✓ built in 10ms
vite v4.4.7 building for production...
transforming (1) index.htmlBrowserslist: caniuse-lite is outdated. Please run:
  npx update-browserslist-db@latest
  Why you should do it regularly: https://github.com/browserslist/update-db#readme
transforming (17) src/components/menu.tsx
./pattern.png referenced in /home/zxp-endeavouros/.cache/paru/clone/salvage-git/src/salvage.git/src/renderer/src/styles/index.css didn't resolve at build time, it will remain unchanged to be resolved at runtime
✓ 1292 modules transformed.
../../out/renderer/index.html                   0.52 kB
../../out/renderer/assets/index-45313666.css   22.62 kB
../../out/renderer/assets/index-d22367f1.js   537.14 kB
✓ built in 1.85s
  • electron-builder  version=23.6.0 os=6.6.16-1-lts
  • loaded configuration  file=/home/zxp-endeavouros/.cache/paru/clone/salvage-git/src/salvage.git/electron-builder.yml
  • writing effective config  file=dist/builder-effective-config.yaml
  • skipped dependencies rebuild  reason=npmRebuild is set to false
  • packaging       platform=linux arch=x64 electron=24.6.5 appOutDir=dist/linux-unpacked
  • downloading     url=https://github.com/electron/electron/releases/download/v24.6.5/electron-v24.6.5-linux-x64.zip size=95 MB parts=8
  • downloaded      url=https://github.com/electron/electron/releases/download/v24.6.5/electron-v24.6.5-linux-x64.zip duration=7.164s
  • building        target=AppImage arch=x64 file=dist/salvage-1.5.3.AppImage
  • downloading     url=https://github.com/electron-userland/electron-builder-binaries/releases/download/appimage-12.0.1/appimage-12.0.1.7z size=1.6 MB parts=1
  • downloaded      url=https://github.com/electron-userland/electron-builder-binaries/releases/download/appimage-12.0.1/appimage-12.0.1.7z duration=5.79s
```
### Error logs
```
$ ./salvage 
(node:88155) UnhandledPromiseRejectionWarning: TypeError: windowInstance.hookWindowMessage is not a function
    at createWindow (/home/zxp-endeavouros/.cache/paru/clone/salvage-git/src/salvage.git/dist/linux-unpacked/resources/app.asar/out/main/index.js:258:18)
    at createWindowAndTray (/home/zxp-endeavouros/.cache/paru/clone/salvage-git/src/salvage.git/dist/linux-unpacked/resources/app.asar/out/main/index.js:221:3)
    at /home/zxp-endeavouros/.cache/paru/clone/salvage-git/src/salvage.git/dist/linux-unpacked/resources/app.asar/out/main/index.js:273:76
(Use `salvage --trace-warnings ...` to show where the warning was created)
(node:88155) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 1)
Initial scan complete. Ready for changes
```
```

**Comment by zxp19821005** [comment]
I've ported it to ArchLinux AUR: https://aur.archlinux.org/packages/salvage-git

**Comment by RenanSui** [comment]
Issue (maybe) addressed on version 2.0.0. I don't have a Linux OS right now to test, but Tauri is compatible with Linux I think.
